Submitty/Submitty
# [Feature:Submission] Timed gradeable popup submit

### What is the current behavior?
closes #4596 

### What is the new behavior?
For timed gradeables, a pop up will appear in the upper right corner of the screen encouraging students to submit so they can save their progress. The popup has 3 buttons, submit, close and don't remind me again. The popup appears every time a set interval is reached, as of now the interval is either 20 percent of the total allowed time or 10 minutes, whichever is greater. The popup will never appear for gradeables with a time limit of less than 25 minutes. The popup will not appear if there is no new work to submit.

### Other information?
To test:

To manually make a gradeable a timed gradeable, go to the "gradeable" table in the database and set the allowed_minutes column for a notebook gradeable.

NOTE allowed minutes must be over 25 or the popup will never show up

1. Setup a timed notebook gradeable
2. attempt to complete the gradeable and view the popup.
3. for testing purposes, change the interval in notebook.twig to a much shorter amount of time so you don't have to wait 20 minutes to see it.
4. use all of the buttons
